better explanation of what aliases to pick (#97)
run-converge/docs-mintlify: sources/concepts/aliases.mdx

</CodeGroup>
</Accordion>

### What aliases should I pass?

> The exact aliases you choose depends on your specific customer journey, commonly used aliases are:

Reviewer: ```suggestion
The exact aliases you choose depend on your specific customer journey, commonly used aliases are:
```

- **Cart Token** (e.g. `urn:cart_token:123xyz`): can be used to stitch a checkout on a different domain with the storefront session; or to stitch a server-side purchase to a website session.
- **Checkout Token** (e.g. `urn:checkout_token:456abc`): can be used to stitch a checkout on a different domain with the storefront session; or to stitch a server-side purchase to a website session.

> - **Email** (e.g. `urn:email:john.smith@apple.com`): can be used to stitch a server-side order to a checkout session

Reviewer 2: user ID is also a good example

> There are two main guidelines for setting up aliases:

Reviewer 2: ```suggestion
There are two main guidelines for choosing aliases:
```

1. **Aim for redundancy**: Client-side events have no guarantee of being tracked, meaning that if the stitching of your customer journey depends on a single browser event and this browser event is missed, your server-side conversion will not be stitched correctly.
2. **As early in the customer journey as possible**: If your stitching alias is too late in the journey, then you run the risk that the server-side `Placed Order`-event is received in Converge, before the client-side event with the stitching alias has been received. As such a `cart_token` you can pick up early in the journey is a much more reliable alias than an `email` property you receive very late in the journey.

> 

Reviewer 2: we can add a separate section for implementation guidelines, e.g. setting the alias whenever it is available, not only when it become available, sanitizing the alias (email -> lowercase, stripped), double checking the format matches, etc.

Author: @thomashlvt agreed that this would be helpful, let's merge this as-is and start adding those above every time issues pop up.

> #### Example 

Reviewer 2: Common pitfalls could also be a useful section. E.g. explain why order ID is a bad idea for an alias

If you have a customer journey that looks like this: 

```mermaid
flowchart LR
  id1("\`*Browser* \n Viewed Page\`") --> id2("\`*Browser* \n Added To Cart \`") --> id3("\`*Browser* \n Started Checkout \`") --> id4("\`*Browser* \n Added Payment Info \`") --> id5("\`*Server* \n Placed Order \`")
```

> Then an optimal stitching setup would look something like:

Reviewer: ```suggestion
Then an optimal stitching setup would look something like this:
```

```mermaid
flowchart LR
  id1("\`*Browser* \n Viewed Page\`") --> id2("\`*Browser* \n Added To Cart \n **alias=cart_token_xyz** \`") --> id3("\`*Browser* \n Started Checkout \n **alias=checkout_token_123**  \`") --> id4("\`*Browser* \n Added Payment Info \n **alias=john@smith.com**  \`") --> id5("\`*Server* \n Placed Order \n **alias=cart_token_xyz** \n **alias=checkout_token_123** \n **alias=john@smith.com** \`")
```

Note that:

1. There are multiple stitching identifiers on the browser events in the customer journey to stitch the server-side `Placed Order` to the corresponding website session.

> 2. There are stitching identifiers available early in the customer journey, making sure that a client-side events with relevant aliases will arrive before the server-side `Placed Order`-event.

Reviewer: ```suggestion
2. Stitching identifiers are available early in the customer journey, ensuring that client-side events with relevant aliases will arrive before the server-side `Placed Order` event.
```
